timeout/timeout.py

```diff
@@ -57,27 +59,49 @@ async def report_handler(self, ctx: commands.Context, user: discord.Member, acti
         # Send embed
         await log_channel.send(embed=embed)
 
-    async def timeout_add(self, ctx: commands.Context, user: discord.Member, reason: str, timeout_role: discord.Role):
+    async def timeout_add(self, ctx: commands.Context, user: discord.Member, reason: str, timeout_role: discord.Role, timeout_roleset: list[discord.Role]):
         """Retrieve and save user's roles, then add user to timeout"""
+        # Catch users already holding timeout role.
+        # This could be caused by an error in this cog's logic or,
+        # more likely, someone manually adding the user to the role.
+        if timeout_role in user.roles:
+            await ctx.send("Something went wrong! Is the user already in timeout? Please check the console for more information.")
+            log.warning(
+                f"Something went wrong while trying to add user {self.target} to timeout.\n" +
+                f"Current roles: {user.roles}\n" +
+                f"Attempted new roles: {timeout_roleset}"
+            )
+            return
+
         # Store the user's current roles
         user_roles = []
         for role in user.roles:
             user_roles.append(role.id)
```

Comment:
```suggestion
        user_roles = [r.id for r in user.roles]
```
It just makes sense 🤷‍♀️